### DISCUSSION TITLE
Volume: step for up/down volume + OSD volume display

### DISCUSSION THREAD

**Discussion by verbal666** [smplayer-dev/smplayer#987] [Q&A]
Hi.
SM is great, very user friendly, and without too many options to look for, minimal.
BUT, i always had some question about

1. How to get rid of volume step when managing volume? I want a -+5 step, but can't get it. I also edit the "min_step=" in .ini file, but seems it's not a real integer step for it. If i set "min_step=5" step is not +-5. Now i set "min_step=1" and it's good to me. But, it's not +-5!!!
2. Volume OSD display is different from Windowed or Fullscreen UI, when PLAY IS ON, in PAUSE MODE is equal; when in Windowed, playing, i see 100 and in Fullscreen it display something like 66 (seems a % comparition to real 100 of the total sw boost requested, read next). I set also "Use sw volume contol" to 125 to manage and gain some volume boost. Without "sw volume control" display is equal 1:1 .

Thanks.

**Comment by Raza-0007** [comment]
I believe both your issues are caused by `use software volume control` option, probably a bug in its implementation. If you uncheck that option, you can then set any volume step size, and the volume step size remains constant in windowed and full-screen mode.

I personally don't use or recommend using `software volume control`, as your sound card mixer will always provide a superior audio output. Software volume control should only be used if you do not have proper audio drivers, or your audio drivers lack the ability to control audio mixer/equalizer settings etc. 

~Raza

**Reply by verbal666** [reply]
I know it's ```use software volume control``` the issue.

The question is WHY with this feature volume OSD is not precise and volume step also.
I use software volume control since sometimes, with 100% volume, A/V files are still too low (i have a 32" with 2 Logitech speakers right next to it, L and R, and i use it from 5mt to follow my media), and i don't want to use the "**boost**" of the Realtek driver, since it turn up volume to extreme (with some apps at volume 1 i hear it too loud...

Anyway, it's not a great problem. I'll see if they fix.

Thanks 👍
